openedx/core/djangoapps/edx_global_analytics/tasks.py

```diff
@@ -0,0 +1,83 @@
+"""
+This file contains periodic tasks for global_statistics, which will collect data about Open eDX users
+and send this data to appropriate service for further processing.
+"""
+
+import json
+import requests
+
+from celery.task import task
+
+from django.conf import settings
+from django.contrib.sites.models import Site
+from django.core.exceptions import ObjectDoesNotExist
+from xmodule.modulestore.django import modulestore
+from student.models import UserProfile
+from .models import TokenStorage
+
+
+@task
+def count_data():
+    """
+    Periodic task function that gathers information about the students amount,
+    geographical coordinates of the platform, courses amount and
+    makes a POST request with the data to the appropriate service.
+    """
+
+    # OLGA settings
+    olga_settings = settings.ENV_TOKENS.get('OPENEDX_LEARNERS_GLOBAL_ANALYTICS')
+
+    # Get IP address of the platform and convert it to latitude, longitude.
+    ip_data = requests.get('http://freegeoip.net/json')
+    ip_data_json = json.loads(ip_data.text)
+    platform_latitude, platform_longitude = olga_settings.get("PLATFORM_LATITUDE"), olga_settings.get("PLATFORM_LONGITUDE")
+   
+    if platform_latitude and platform_longitude:
+        latitude, longitude = platform_latitude, platform_longitude
+    else:
+        latitude, longitude = ip_data_json['latitude'], ip_data_json['longitude']
+
+    # Get students amount within current platform.
+    students_amount = UserProfile.objects.count()
+
+    # Get courses amount within current platform.
+    courses_amount = len(modulestore().get_courses())
+
+    # Secret token to authorize our platform on remote server.
+    try:
+        token_object = TokenStorage.objects.get(pk=1)
```

Comment:
Do I need post comment for all parts of code I want to change in future, although it may be 25% or smth. like that of total current code?